### DISCUSSION TITLE
Using group notifications

### DISCUSSION THREAD

**Discussion by philitell** [781flyingdutchman/background_downloader#418] [Q&A]
Hi, i'm trying to use notifications for groups.
I have a large number of DownloadTasks within a group (e.g. 80 tasks). Those tasks are all enqueued and i would like to show a notification that says e.g. "25 of 80" with a progress-indicator.

The problem is that i'm seeing an notification that gets updated with every status update (when a Task is started and completed). So the notifications shows "0 of 1" - then gets updated to "1 of " - then when the next file is downloaded the notification is updated and shows "0 of 1" again. This happens for all 80 files.

What i'm doing wrong?

```
// here is the general initialization/configuration
     await FileDownloader().ready;
      final configResult = await FileDownloader().configure(
        globalConfig: [
          (Config.requestTimeout, const Duration(seconds: 100)),
          (Config.checkAvailableSpace, 500),
          // NOTE: queue and sets the constraints maxConcurrent to 3, maxConcurrentByHost to 2,
          // and maxConcurrentByGroup to 1.
          (Config.holdingQueue, (3, 2, 1)),
        ],
        androidConfig: [
          (Config.useCacheDir, Config.whenAble),
        ],
        iOSConfig: [
          (Config.localize, {"Cancel": S.current.stopIt}),
        ],
      );

      // Register an event listener to process status and progress updates
      _ref.read(fileTransferTaskStatusUpdateProvider);

      repositoryLogger.info("Start persistent task tracking");
      await FileDownloader().trackTasks();

      repositoryLogger.info("Check for events that happened while the app was in the background");
      await FileDownloader().resumeFromBackground();
....

// this is my notification configruation which is created right before the tasks of the current group are enqueued (i'm NOT using downloadBatch because is need to to persistent task tracking as well)
      FileDownloader().configureNotificationForGroup(
        groupIdentifier,
        running: const TaskNotification("{numFinished} von {numTotal}", "Fortschritt: {progress}"),
        complete: const TaskNotification("Abgeschlossen", "{numTotal} Dateien heruntergeladen"),
        error: const TaskNotification("Fehler", "{numFailed}/{numTotal} fehlgeschlagen"),
        progressBar: true,
        groupNotificationId: "notification_$groupIdentifier",
      );
```

**Comment by philitell** [comment]
It seems to be related with the configuration of the holdingQueue

**Comment by 781flyingdutchman** [comment]
Which platform (Android, iOS, Desktop) is this happening on, and does it happen on the other(s)?  Can you confirm that if you don't use a holding queue the groupnotification works as intended?

**Reply by philitell** [reply]
The project is in development stage - so the issue occurred on android emulator (Android Studio).

And yes i can confirm that when i'm not using holding queue it works as intendend. Furthermore if i'm using holding queue and configure e.g maxConcurrentByHost = 3 then the notification show progress 0/3; 1/3; 2/3; 3/3 (for the first 3 files) and continues with 0/3; 1/3... for the next three files. Meaning that the total number of tasks is divided into chucks of size == maxConcurrentByHost.

**Reply by 781flyingdutchman** [reply]
Thanks, and yes I can see where this goes wrong, as the number of outstanding jobs is based on those enqueued with the WorkManager, not with the HoldingQueue.  I created an issue for this #420 to make it easier to track

**Comment by 781flyingdutchman** [comment]
Per issue #420, published in V8.9.0
